package.json

```diff
@@ -115,6 +115,7 @@
     "@babel/preset-env": "7.16.11",
     "@babel/preset-react": "7.16.7",
     "@babel/register": "7.17.7",
+    "@bloomberg/record-tuple-polyfill": "^0.0.4",
```

Comment:
Do we really need a polyfill to add this behaviour? If we want to use tuple types here, I think I'd rather just use a plain array and trust that we don't modify the result rather than introduce an experimental polyfill for it. We're also having to work around the fact that it doesn't natively support TypeScript in quite a few places, and it seems like TypeScript may not natively support it for some time even once the proposal reaches stage 3.

All-in-all, I'd rather we not use this until it becomes more standardized. The rest of the logic of returning `[href, type]` and passing that around still looks fine to me though as long as we're careful to use a shallow equality check instead of `===`

Reply:
Oh, I see. I didn't realize that Maps needed the keys to be checkable by `===` to work. I knew one of the benefits of Map is that it supports non-string keys, but I don't think I've ever ended up actually using that.

In that case, I think these are fine even if they don't have proper support yet. I've definitely wanted something like this in the past and had to settle for encoding things as strings, but this is definitely nicer than that.